content/api/events.apib

```diff
@@ -0,0 +1,269 @@
+
+To start a paging request, add the query parameter `?cursor=initial` to your Events API request. This will fetch the first page of results. If cursor is not provided, then the default value of `initial` is applied.
+In the response body, there are three main keys: `results`, `total_count` and `links`. `results` is an array which contains the page of JSON events that matched the request,
+`total_count` is an integer that describes the total number of JSON events that matched your request, and `links` is an object which provides a way to request the next page of events, if there is more than one page.
+
+Once you have the response, you then can use `links` to retrieve the URL for your next page, which will be prefilled with the appropriate value for the `cursor` query parameter,
+as well as any other filters you supplied with the original request.
+
+In the following examples, to retrieve the first page, we use the `initial` value for the `cursor` query parameter.  the `cursor` value from
+the `next` key in the `links` object, and use that to make the request for the second page. This process would then be repeated until all results have been paged through,
+at which point the `links` object will not have a `next` key, and no further additional requests are necessary.
+
+If your result set is less than one page of data, then the `links` object will not have a `next` key.
+
+<Banner status="info">Pagination requests count towards the number of requests allowed by rate limiting, the same as non-paginated requests.</Banner>
+
+### Retrieve First Page [GET /v1/events/message{?per_page,cursor}]
+
++ Parameters
+    + per_page: `1000` (string, optional) 
+    + cursor: `initial` (string, optional)
+
++ Response
+```
+    {
+      "results": [
+        ...
+      ],
+      "total_count": 20000,
+      "links": {
+        "next": "/api/v1/events/message?events=delivery&per_page=1000&cursor=WycyMDE4LTExLTA1VDIyOjQ1OjM5LjAwMFonLCAnc3BjLTM4MTQ1MjY3MjMyNTA2NTEwJ10="
+      }
+    }
+```
+
+### Retrieve Subsequent Page [GET /v1/events/message{?per_page,cursor}]
++ Parameters
+    + per_page: `1000` (string, optional) 
+    + cursor: `WycyMDE4LTExLTA1VDIyOjQ1OjM5LjAwMFonLCAnc3BjLTM4MTQ1MjY3MjMyNTA2NTEwJ10=` (string, optional)
+
++ Response
+```
+    {
+      "results": [
+        ...
+      ],
+      "total_count": 20000,
+      "links": {
+        "next": "/api/v1/events/message?events=delivery&per_page=1000&cursor=WycyMDE4LTExLTA0VDIyOjQ1OjM5LjAwMFonLCAndWF0LTM4MTQ1MjE3MjUyNTY2NTIwJ10="
+      }
+    }
+```
+
+## Search [/v1/events]
+
+### Search for Message Events [GET /v1/events/message{?ab_tests,ab_test_version,bounce_classes,campaigns,cursor,delimiter,events,from,from_addresses,ip_pools,message_ids,per_page,reason,recipients,recipient_domains,subject,sending_domains,sending_ips,subaccounts,templates,timezone,to,transmission_ids}]
+
+Returns a list of message events that matched the filtered search. The response is sorted by descending timestamp.
```

Comment:
Add new optional `sort` and `order` query params.
sort value can be be any of the following: 
```
  'timestamp',
  'recipients',
  'campaigns',
  'templates',
  'message_ids',
  'transmission_ids',
  'from_addresses',
  'bounce_classes',
  'reason',
  'subject',
  'sending_domains',
  'sending_ips',
  'ip_pools',
  'ab_tests'
```
with `timestamp` being the default

`order` can be either `asc` or `desc`, with `desc` being the default.